### PR TITLE
DTFS 4693: Unissued facilities report

### PR DESCRIPTION
**Pull request by IgnatG**
- Added the new `Reports` page
- Added the page for `Unissued facilities reports`
- Download Unissued facilities CTA
- fixed minor typos

### CODE REVIEW

**Review by ttbarnes — COMMENTED** [review]
There's empty `journeys/reports/reports.spec.js ` file, do we need it?

File: e2e-tests/portal/cypress/support/portal-api/api.js [hunk 1]
```diff
@@ -190,6 +190,19 @@ module.exports.insertGefApplication = (deal, token) => cy.request({
   return resp.body;
 });
 
+exports.setGefSubmissionType = (dealId, payload, token) => cy.request({
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal-api/api.js:193]: Isn't this a general "update GEF" endpoint? Could be `updateGefDeal` for future use?

File: portal/server/controllers/dashboard/reports.controller.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  return res.redirect('/not-found');
+};
+
+exports.downloadUnissuedFacilitiesReports = async (req, res) => {
+  const { userToken } = req.session;
+  const facilities = await api.getUnissuedFacilitiesReports(userToken);
+
+  // not all columns are needed, in which case, the array below will specify
+  // the properties that we want to add to the CSV file and the label for them
```
Comment (ttbarnes) [inline portal/server/controllers/dashboard/reports.controller.js:40]: what does the first part mean please, `not all columns are needed`, not sure why/the meaning - looks like all the columns are needed? 🙃

File: e2e-tests/portal/cypress/integration/journeys/reports/unissued-facilities-reports.spec.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      cy.get('@row1').find('[data-cy="facility__bankRef"]').should('contain', 'Draft GEF');
+      cy.get('@row1').find('[data-cy="facility__product"]').should('contain', 'GEF');
+      cy.get('@row1').find('[data-cy="facility__facilityId"]').should('contain', '00000001');
+      cy.get('@row1').find('[data-cy="facility__companyName"]').should('contain', 'Delta');
+      cy.get('@row1').find('[data-cy="facility__facilityValue"]').should('contain', 'GBP 123');
+      cy.get('@row1').find('[data-cy="facility__deadlineForIssuing"]').should('contain', '20 Apr 2022');
+      cy.get('@row1').find('[data-cy="facility__daysLeftToIssue"]').should('contain', '90 days left');
+    });
+  });
+});
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/integration/journeys/reports/unissued-facilities-reports.spec.js:63]: I can't see an e2e test for the verious template conditions ie if days left to issue is 15 / less than 0 etc. Can we add them in?

File: e2e-tests/portal/cypress/support/portal-api/api.js (hunk 1 above)
Comment (IgnatG) [reply e2e-tests/portal/cypress/support/portal-api/api.js:193]: good shout - it's updating the entire deal. I've changed this

File: e2e-tests/portal/cypress/integration/journeys/reports/unissued-facilities-reports.spec.js (hunk 3 above)
Comment (IgnatG) [reply e2e-tests/portal/cypress/integration/journeys/reports/unissued-facilities-reports.spec.js:110]: my bad - the are in now

File: portal-api/src/v1/controllers/reports/reports.controller.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    },
+    // sort based on the submissionDate in ASC order - most urgent item should be first
+    { $sort: { submissionDate: 1 } },
+  ]).toArray();
+
+  return facilities;
+};
+
+// this function is used to build the "Unissued facilities" page for on Reports
+exports.findUnissuedFacilitiesReports = async (req, res) => {
```
Comment (abhi-markan) [inline portal-api/src/v1/controllers/reports/reports.controller.js:83]: Wrap it in a try-catch block please.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: portal-api/src/v1/controllers/reports/reports.controller.js (hunk 4 above)
Comment (IgnatG) [reply portal-api/src/v1/controllers/reports/reports.controller.js:83]: I've added this now

**Review by abhi-markan — APPROVED** [review]

**Review by abhi-markan — APPROVED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**